## CSMODEL MP PHASE 1
__Group 5 Members:__
* Joshua Esleta
* John Liong
* Mark Musngi
* Justine Valdes




# Import Libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy.stats import norm
from scipy.stats import ttest_ind

## DataFrame

Variables and Dictionaries are in the file "fies_2012_v1_metadata(dictionary)"

In [2]:
df = pd.read_csv("FIES PUF 2012 Vol.1.csv")
df

W_REGN       W_OID  W_SHSN  W_HCN  URB   RSTR     PSU  BWEIGHT  \
0          14   101001000       2     25    2  21100  415052   138.25   
1          14   101001000       3     43    2  21100  415052   138.25   
2          14   101001000       4     62    2  21100  415052   138.25   
3          14   101001000       5     79    2  21100  415052   138.25   
4          14   101001000      10    165    2  21100  415052   138.25   
...       ...         ...     ...    ...  ...    ...     ...      ...   
40166      12  9804029001      18    568    1  22000  114062   271.25   
40167      12  9804035000       1     25    2  22000  414067   271.25   
40168      12  9804035000       2     51    2  22000  414067   271.25   
40169      12  9804035000       3     75    2  22000  414067   271.25   
40170      12  9804035000       4    102    2  22000  414067   271.25   

          RFACT  FSIZE  ...  PC_QTY  OVEN_QTY  MOTOR_BANCA_QTY  \
0      200.6576    3.0  ...      01        01                    
1      200.6576   12.5  ...                01                    
2      200.6576    2.0  ...                01                    
3      200.6576    4.0  ...                01                    
4      200.6576    5.0  ...                                      
...         ...    ...  ...     ...       ...              ...   
40166  963.2264    5.0  ...      00        00               00   
40167  588.6253    9.0  ...      00        01               00   
40168  588.6253    6.0  ...      00        00               00   
40169  588.6253    5.0  ...      00        01               00   
40170  588.6253    7.0  ...      00        00               00   

       MOTORCYCLE_QTY   POP_ADJ      PCINC  NATPC  NATDC  REGDC  REGPC  
0                      0.946172  108417.00      9      8      8      9  
1                  01  0.946172   30631.60      5      9      9      4  
2                      0.946172   86992.50      9      6      6      8  
3                      0.946172   43325.75      6      6      6      6  
4                  01  0.946172   37481.80      6      6      6      5  
...               ...       ...        ...    ...    ...    ...    ...  
40166              00  0.902863   30101.60      5      5      7      7  
40167              00  0.902863   14368.89      1      5      7      3  
40168              00  0.902863   19137.33      3      4      6      4  
40169              00  0.902863   30985.00      5      6      7      7  
40170              00  0.902863   15738.00      2      4      6      3  

[40171 rows x 119 columns]

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40171 entries, 0 to 40170
Columns: 119 entries, W_REGN to REGPC
dtypes: float64(5), int64(92), object(22)
memory usage: 36.5+ MB


# Dataset Description
## Family Income and Expenditure Survey 2012
 
### __Brief Description__
The 2012 Family Income and Expenditure Survey (FIES) adopted the 2009 Philippine Classification of Individual Consumption According to Purpose (PCOICOP). The 2009 PCOICOP is the first standard classification of individual consumption expenditure in the country prepared by the National Statistical Coordination Board (NSCB) in collaboration with concerned agencies in the Philippine Statistical System (PSS). The 2009 PCOICOP was patterned after the 1999 Classification of Individual Consumption According to Purpose (COICOP) issued by the United Nations Statistics Division (UNSD). The 2012 FIES is a sample survey designed to provide income and expenditure data for the country and its seventeen (17) regions. These data are the basic sources of information on human development and the basis of the policies and programs of the government to improve the socio-economic conditions of the Filipino people. 

### __Description of the Collection Process__

 The survey was conducted in two separate rounds using separate questionnaires with the same set of questions. The first was conducted in July 2012 and the second was in January 2013. For each round, the reference period was the half-year period preceding the survey. The primary strata was the 17 regions of the country and used the 2003 Master Sample (MS) as its sampling framework, which was also based on the 2000 Census of Population and Housing (CPH). With the 17 regions, within each one, another level of stratification was done using geographic subdivisions such as provinces. Highly Urbanized Cities (HUCs) and Independent Component Cities (ICCs). After this, Primary Sampling Units (PSUs) were selected by using the probability proportional to size. With this, larger PSUs had a higher chance of selection in the random sampling. After, Enumeration Areas (EAs) were chosen within each sampled PSU. Overall. the collection procedure involved a national sample of approximately 50,000 households to ensure coverage of both national and regional levels.

### __Structure of the dataset__

#### What does each row and column represent?

* Each row represents a household and each column represents a different variable about that household that was collected in the survey

#### How many observations are there in the dataset?

* There are 40,171 observations in the dataset

#### How many variables are there in the dataset?

* There are 119 variables in the dataset

### __Variables in the dataset__

<details>
<summary>See full list of variables and their description</summary>

  __w_regn__

  * Region the household belongs to

  __w_oid__

  * Other ID for every sample household

  __w_shsn__

  * Sequential number of sample households in the primary sampling unit (PSU)/EA

  __w_hcn__

  * Household Control Number assigned to each household in the barangay/EA

  __psu__

  * Primary Sampling Unit (Recode)

  __urb__

  * Urban or Rural indicator (1 for Urban, 2 for Rural)

  __rstr__

  * Stratum (Recode) based on geographic and socio-economic variables

  __bweight__

  * Base weight assigned based on the inverse of the selection probability

  __rfact__

  * Final weight calculated with adjustments for non-response and population

  __fsize__

  * Family size, representing the number of members in the family

  __agri_sal__

  * Salaries and wages received from employment in agricultural sectors

  __nonagri_sal__

  * Salaries and wages received from employment in non-agricultural sectors

  __wages__

  * Total salaries and wages from both agricultural and non-agricultural activities

  __netshare__

  * Net share of crops, fruits, and vegetables produced or livestock raised by other households

  __cash_abroad__

  * Cash receipts, support, relief, and other forms of assistance from abroad

  __cash_domestic__

  * Cash receipts, support, assistance, and relief from domestic sources

  __rentals_rec__

  * Rentals received from non-agricultural lands, buildings, and other properties

  __interest__

  * Interest received from bank deposits and loans extended to other families

  __pension__

  * Pension and retirement benefits received

  __dividends__

  * Dividends from investments such as stocks and bonds

  __other_source__

  * Other sources of income not elsewhere classified

  __net_receipt__

  * Total net receipts from family sustenance activities

  __regft__

  * Total received as gifts

  __net_cfg__

  * Net income from crop farming and gardening

  __net_lpr__

  * Net income from livestock and poultry raising

  __net_fish__

  * Net income from fishing

  __net_for__

  * Net income from forestry and hunting

  __net_ret__

  * Net income from wholesale and retail trade

  __net_mfg__

  * Net income from manufacturing

  __net_com__

  * Net income from community, social, recreational, and personal services

  __net_trans__

  * Net income from transportation, storage, and communication services

  __net_min__

  * Net income from mining and quarrying

  __net_cons__

  * Net income from construction

  __net_nec__

  * Net income from other entrepreneurial activities not elsewhere classified

  __eainc__

  * Total income from entrepreneurial activities

  __toinc__

  * Total income

  __losses__

  * Losses from entrepreneurial activities

  __t_bread__

  * Total expenditure on bread and cereal

  __t_meat__

  * Total expenditure on meat

  __t_fish__

  * Total expenditure on fish and seafood

  __t_milk__

  * Total expenditure on milk, cheese, and eggs

  __t_oil__

  * Total expenditure on oils and fats

  __t_fruit__

  * Total expenditure on fruits

  __t_veg__

  * Total expenditure on vegetables

  __t_food_nec__

  * Total expenditure on food not elsewhere classified

  __t_sugar__

  * Total expenditure on sugar, jam, honey, chocolate, and confectionery

  __t_coffee__

  * Total expenditure on coffee, cocoa, and tea

  __t_mineral__

  * Total expenditure on mineral water, soft drinks, fruit, and vegetable juices

  __t_alcohol__

  * Total expenditure on alcoholic beverages

  __t_tobacco__

  * Total expenditure on tobacco

  __t_other_veg__

  * Total expenditure on other vegetable-based products

  __t_food_home__

  * Total food expenditure at home

  __t_food_outside__

  * Total food expenditure outside the home

  __t_food__

  * Total food expenditure

  __t_cloth__

  * Total expenditure on clothing and footwear

  __t_furnishings__

  * Total expenditure on furnishings, household equipment, and routine maintenance

  __t_health__

  * Total expenditure on health

  __t_housing__

  * Total expenditure on housing and utilities

  __t_actrent__

  * Actual rent paid for housing

  __t_rentval__

  * Rental value of the house

  __t_imputed_rent__

  * Imputed rental value of owned house

  __t_transp__

  * Total expenditure on transportation

  __t_comm__

  * Total expenditure on communication

  __t_recreation__

  * Total expenditure on recreation and culture

  __t_education__

  * Total expenditure on education

  __t_misc__

  * Total expenditure on miscellaneous goods and services

  __t_other_exp__

  * Total other expenditure including value consumed and losses

  __t_nfood__

  * Total non-food expenditure

  __t_totex__

  * Total expenditure

  __t_totdis__

  * Total disbursements

  __t_other_receipts__

  * Total other receipts

  __t_torec__

  * Total receipts

  __food_accommodation__

  * Food regularly consumed outside the home in accommodation services

  __sex__

  * Sex of the household head (2nd visit only)

  __age__

  * Age of the household head (2nd visit only)

  __ms__

  * Marital status of the household head (2nd visit only)

  __hgc__

  * Highest grade completed by the household head (2nd visit only)

  __job__

  * Job or business indicator for the household head (2nd visit only)

  __occup__

  * Occupation of the household head (2nd visit only)

  __kb__

  * Kind of business or industry of the household head (2nd visit only)

  __cw__

  * Class of worker of the household head (2nd visit only)

  __hhtype__

  * Type of household (2nd visit only)

  __members__

  * Total number of family members (2nd visit only)

  __ageless5__

  * Members with age less than 5 years (2nd visit only)

  __age5_17__

  * Members with age between 5 and 17 years (2nd visit only)

  __employed_pay__

  * Total number of family members employed for pay (2nd visit only)

  __employed_profit__

  * Total number of family members employed for profit (2nd visit only)

  __spouse_emp__

  * Indicator if the spouse has a job/business (2nd visit only)

  __bldg_type__

  * Type of building/house (2nd visit only)

  __roof__

  * Type of roof (2nd visit only)

  __walls__

  * Type of walls (2nd visit only)

  __tenure__

  * Tenure status of the housing unit (2nd visit only)

  __hse_alterations__

  * Whether alterations or additions to the house were made (2nd visit only)

  __toilet__

  * Type of toilet facilities (2nd visit only)

  __electricity__

  * Electricity indicator (2nd visit only)

  __water__

  * Main source of water supply (2nd visit only)

  __distance__

  * Distance of water source from the house (2nd visit only)

  __radio_qty__

  * Number of radios (2nd visit only)

  __tv_qty__

  * Number of televisions (2nd visit only)

  __cd_qty__

  * Number of CD/VCD/DVD players (2nd visit only)

  __stereo_qty__

  * Number of component/stereo sets (2nd visit only)

  __ref_qty__

  * Number of refrigerators/freezers (2nd visit only)

  __wash_qty__

  * Number of washing machines (2nd visit only)

  __aircon_qty__

  * Number of air conditioners (2nd visit only)

  __car_qty__

  * Number of cars, jeeps, or vans (2nd visit only)

  __landline_qty__

  * Number of landline/wireless telephones (2nd visit only)

  __cellphone_qty__

  * Number of cellular phones (2nd visit only)

  __pc_qty__

  * Number of personal computers (2nd visit only)

  __oven_qty__

  * Number of stoves with ovens or gas ranges (2nd visit only)

  __motor_banca_qty__

  * Number of motorized bancas (2nd visit only)

  __motorcycle_qty__

  * Number of motorcycles/tricycles (2nd visit only)

  __pop_adj__

  * Population adjustment factor

  __pcinc__

  * Per capita income

  __natpc__

  * National per capita income decile

  __natdc__

  * National income decile

  __regdc__

  * Regional income decile

  __regpc__

  * Regional per capita income decile

</details>


# Data Cleaning

### Variables to be cleaned
* W_REGN
* NONAGRI_SAL
* EMPLOYED_PAY
* OCCUP
* JOB

In [4]:
# Select only wanted variables
selected_df = df[["W_REGN","W_OID","W_SHSN","W_HCN","NONAGRI_SAL","EMPLOYED_PAY","OCCUP","JOB"]]

In [5]:
selected_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40171 entries, 0 to 40170
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   W_REGN        40171 non-null  int64 
 1   W_OID         40171 non-null  int64 
 2   W_SHSN        40171 non-null  int64 
 3   W_HCN         40171 non-null  int64 
 4   NONAGRI_SAL   40171 non-null  int64 
 5   EMPLOYED_PAY  40171 non-null  object
 6   OCCUP         40171 non-null  object
 7   JOB           40171 non-null  int64 
dtypes: int64(6), object(2)
memory usage: 2.5+ MB


## Dropped Duplicates


In [6]:
selected_df = selected_df.drop_duplicates()

In [7]:
selected_df = selected_df.dropna(subset=["OCCUP"])
selected_df

W_REGN       W_OID  W_SHSN  W_HCN  NONAGRI_SAL EMPLOYED_PAY OCCUP  JOB
0          14   101001000       2     25            0                       2
1          14   101001000       3     43            0               1314    1
2          14   101001000       4     62            0                       2
3          14   101001000       5     79            0               6111    1
4          14   101001000      10    165        32000           01  5220    1
...       ...         ...     ...    ...          ...          ...   ...  ...
40166      12  9804029001      18    568            0           00  1314    1
40167      12  9804035000       1     25        50400           01  8321    1
40168      12  9804035000       2     51            0           00  1314    1
40169      12  9804035000       3     75       100761           01  1120    1
40170      12  9804035000       4    102            0           00  6121    1

[40171 rows x 8 columns]

## Drop those who doesnt have a job

In [8]:
selected_df = selected_df[selected_df["JOB"] != 2]
selected_df

W_REGN       W_OID  W_SHSN  W_HCN  NONAGRI_SAL EMPLOYED_PAY OCCUP  JOB
1          14   101001000       3     43            0               1314    1
3          14   101001000       5     79            0               6111    1
4          14   101001000      10    165        32000           01  5220    1
5          14   101001000      14    229       749628           03  6212    1
7          14   101001000      18    295            0           02  6111    1
...       ...         ...     ...    ...          ...          ...   ...  ...
40166      12  9804029001      18    568            0           00  1314    1
40167      12  9804035000       1     25        50400           01  8321    1
40168      12  9804035000       2     51            0           00  1314    1
40169      12  9804035000       3     75       100761           01  1120    1
40170      12  9804035000       4    102            0           00  6121    1

[32797 rows x 8 columns]

## Drop Variable who aren't in agriculture

In [9]:
selected_df = selected_df[selected_df["NONAGRI_SAL"] > 0]
selected_df

W_REGN       W_OID  W_SHSN  W_HCN  NONAGRI_SAL EMPLOYED_PAY OCCUP  JOB
4          14   101001000      10    165        32000           01  5220    1
5          14   101001000      14    229       749628           03  6212    1
11         14   101001000      24    392        82204           01  6111    1
13         14   101002000       2     44         6900           02  5132    1
17         14   101002000       7    162        98100           01  6111    1
...       ...         ...     ...    ...          ...          ...   ...  ...
40163      12  9804029001      13   8004        54300               8321    1
40164      12  9804029001      15   8005        54300           01  1314    1
40165      12  9804029001      17    537        58100           01  8321    1
40167      12  9804035000       1     25        50400           01  8321    1
40169      12  9804035000       3     75       100761           01  1120    1

[21427 rows x 8 columns]

## Convert Employed Pay from string to int

In [10]:
selected_df.loc[:, "EMPLOYED_PAY"] = pd.to_numeric(selected_df["EMPLOYED_PAY"], errors='coerce')
selected_df.loc[:, "EMPLOYED_PAY"] = selected_df["EMPLOYED_PAY"].fillna(0).astype(int)
employed_df = selected_df[["EMPLOYED_PAY"]]
employed_df

/var/folders/2f/2t8tt8fj4sg3t0ljqj26rgpw0000gn/T/ipykernel_79478/2451446224.py:2: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  selected_df.loc[:, "EMPLOYED_PAY"] = selected_df["EMPLOYED_PAY"].fillna(0).astype(int)


EMPLOYED_PAY
4                1
5                3
11               1
13               2
17               1
...            ...
40163            0
40164            1
40165            1
40167            1
40169            1

[21427 rows x 1 columns]

## Drop Rows who has 2 or more people who are working in a household

In [11]:
selected_df = selected_df[selected_df["EMPLOYED_PAY"] == 1]
selected_df

W_REGN       W_OID  W_SHSN  W_HCN  NONAGRI_SAL EMPLOYED_PAY OCCUP  JOB
4          14   101001000      10    165        32000            1  5220    1
11         14   101001000      24    392        82204            1  6111    1
17         14   101002000       7    162        98100            1  6111    1
27         14   101030001       1      1        63246            1  1130    1
33         14   101030001      12    191       293556            1  2331    1
...       ...         ...     ...    ...          ...          ...   ...  ...
40160      12  9804029001       7    213        72000            1  9132    1
40164      12  9804029001      15   8005        54300            1  1314    1
40165      12  9804029001      17    537        58100            1  8321    1
40167      12  9804035000       1     25        50400            1  8321    1
40169      12  9804035000       3     75       100761            1  1120    1

[10066 rows x 8 columns]

## Group Everything by Region

In [12]:
grouped_regions = selected_df.groupby(["W_REGN", "OCCUP"])
grouped_regions = grouped_regions.agg({"NONAGRI_SAL" : "mean"})

grouped_regions

NONAGRI_SAL
W_REGN OCCUP               
1      0930    42127.500000
       1130    62580.000000
       1210    25000.000000
       1227   165000.000000
       1235   334000.000000
...                     ...
42     9311    10800.000000
       9312    58560.000000
       9313    77578.000000
       9322    61066.000000
       9333    79311.428571

[1971 rows x 1 columns]